Fleet API Release stable 2025-03-01

frantran opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [X] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [X] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information
Changes consist of approved ARM changes from:
- https://github.com/Azure/azure-rest-api-specs/pull/32614/
- https://github.com/Azure/azure-rest-api-specs/pull/32628
- https://github.com/Azure/azure-rest-api-specs/pull/32634
- https://github.com/Azure/azure-rest-api-specs/pull/32633

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

frantran committed:
run Copy-ApiVersion.ps1 script for 2025-03-01 api version  (#32514)

* Copy files from preview/2024-05-02-preview

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to stable/2025-03-01

Updated the API version from preview/2024-05-02-preview to stable/2025-03-01.

* add manual readme changes

* run prettier

* run prettier for all

* run pretty on fleets.json

---------

Co-authored-by: frantran <ftran@microsoft.com>

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Microsoft.ContainerService-fleet](https://apiview.dev/Assemblies/Review/c73985c0689e4c56ba378200ebd2a835?revisionId=0c4e54d40f834f819ef30d4f69795183)

mikeharder commented:
/azp run

frantran committed:
fleet autoupgrade GA - add autoupgradepofileid to updaterun (#32633)

* add autoupgradepofileid to updaterun

* generate via tsp

* run generate examples

* run generate examples

* recompile, fix exmamples

* fix examples

* move examples

* fix exmamples

* fix example

* fix examples

---------

Co-authored-by: frantran <ftran@microsoft.com>

audrastump committed:
Fleet status fields (#32614)

* Update version to stable/2025-03-01

Updated the API version from preview/2024-05-02-preview to stable/2025-03-01.

* run prettier

* run prettier for all

* updated

* updated api versions

* added status

* tidied

* tidied

* formatted

* added examples

* restoring readme

* restoring readme

* reformatting

* added visibility to fleet status

---------

Co-authored-by: frantran <ftran@microsoft.com>

frantran committed:
fleet autoupgrade GA - add autoupgradeprofilestatus (#32634)

* add autoupgradeprofilestatus

add tsp files

add v5 version

generate examples

transfer examples

try generate oav

fix generate

transfer examples

fix  gen

fix format

run prettier on examples

manual fix examples

copy examples

fix updateid

fix subid

fix lro header

copy

run pretty

add sub param

copy

add sub param

add sub

reset examples

reset to release

reset examples

* regen tsp

* fix examples

* copy

* regen

* Fix exmaple

* fix more examples

* regen tsp for v3 vs v5 mismatch

---------

Co-authored-by: frantran <ftran@microsoft.com>

frantran committed:
fleet autoupgrade GA - public generate API (#32628)

* add public generate api

* add generation

* regen tsp

* regen examples

* remove header from public spec

* manual changes

* copy

* regen

* add changes

* copy

* run pretty

* change to generateUpdateRuns

* run pretty

* rerun compile

* fix examples

* reset examples to release branch, add profile operation

* fix read visibility, change documentation message

---------

Co-authored-by: frantran <ftran@microsoft.com>

frantran committed:
replace read visibility tag with Lifecycle.Read (#33196)

* remove legacy read tag

* format whitespace

---------

Co-authored-by: frantran <ftran@microsoft.com>

frantran commented:
as a note all these changes were already previously approved by ARM:

- https://github.com/Azure/azure-rest-api-specs/pull/32614
- https://github.com/Azure/azure-rest-api-specs/pull/32628
- https://github.com/Azure/azure-rest-api-specs/pull/32634
- https://github.com/Azure/azure-rest-api-specs/pull/32633

MaryGao commented on specification/containerservice/Fleet.Management/update/autoupgradeprofile.tsp:
```diff
@@ -124,3 +128,59 @@ interface AutoUpgradeProfiles {
+interface AutoUpgradeProfileOperations {
+  #suppress "@azure-tools/typespec-azure-resource-manager/lro-location-header" "use azure-async instead"
+  @useFinalStateVia("azure-async-operation")
+  generateUpdateRun is Azure.ResourceManager.ArmResourceActionAsync<
```
@frantran I'd like to confirm the return type for this operation is `void` or `GenerateResponse`.

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/stable/2025-03-01/fleets.json#L806-L848
